plugins.d/Lets_Encrypt/get_certificate.py

```diff
@@ -73,6 +74,10 @@ def invalid_domains(domains):
                 return ('Error in {}: Domain names must not exceed 254'
                         ' characters'.format(domain))
             for part in domain.split('.'):
+                if not len(domain.split('.')) > 1:
```

Comment:
This check seems a little off

- checks domain segment count, but inside an iteration over segments. If there are 0 segments (which it appears should be an error case) this check will never occur. Best to lift it out of the `for part in domain.split('.')` loop
- instead of `not segment_count > 1`, `segment_count < 2` would be better
- `domain.count('.')` would be a more succinct way to count segments than `len(domain.split('.'))`

Reply:
@NitrogenUA - great idea to add additional domain validation here. However, as @OnGle notes, it should probably be before the `for` loop, instead of inside it.

@OnGle - Thanks! I wasn't aware of `count()` - looks like a very handy string method. And certainly looks like a great way to validate the number of domain segments here.

Reply:
@OnGle thanks for pointing this out while also offering a suggestion! This is a first for me in a number of ways. Admittedly I was feeling too timid, trying to add as little as possible to the detriment of clarity.

Tbh, I couldn't locate a zero-segment scenario you mentioned, outside of us deliberately using `count()` with incorrect delimiter with. But hopefully I understood you correctly and you'll find present implementation satisfactory.